packages/thirdweb/src/extensions/erc4337/account/addSessionKey.ts

```diff
@@ -86,3 +91,84 @@ export function addSessionKey(
 export function isAddSessionKeySupported(availableSelectors: string[]) {
   return isSetPermissionsForSignerSupported(availableSelectors);
 }
+
+/**
+ * Checks if the session key should be updated.
+ * @param currentPermissions - The current permissions of the session key.
+ * @param newPermissions - The new permissions to set for the session key.
+ * @returns A boolean indicating if the session key should be updated.
+ */
+export async function shouldUpdateSessionKey(args: {
```

Comment:
We also check on `startDate` and `isAdmin` values in [our current version of this logic](https://github.com/TreasureProject/tdk-js/blob/main/packages/core/src/connect/session.ts#L101), but not sure if that's important here.

Reply:
start date is a bit weird, decided to leave it alone for this. And yeah admin should be separate i think?